src/main/java/io/strimzi/kafka/metrics/DataPointSnapshotBuilder.java

```diff
@@ -12,20 +12,30 @@
 import io.prometheus.metrics.model.snapshots.Quantiles;
 import io.prometheus.metrics.model.snapshots.SummarySnapshot;
 
+import java.util.logging.Logger;
 /**
  * Builder methods to convert Kafka metrics into Prometheus metrics
  */
 public class DataPointSnapshotBuilder {
 
+    private static final Logger LOG = Logger.getLogger(DataPointSnapshotBuilder.class.getName());
+
     /**
      * Create a datapoint for a {@link InfoSnapshot} metric
      * @param labels The labels associated with the datapoint
      * @param value The value to insert as a label
      * @param metricName The name of the new label
      * @return The {@link InfoSnapshot.InfoDataPointSnapshot} datapoint
      */
+
     public static InfoSnapshot.InfoDataPointSnapshot infoDataPoint(Labels labels, Object value, String metricName) {
-        Labels newLabels = labels.add(PrometheusNaming.sanitizeLabelName(metricName), String.valueOf(value));
+        String sanitizedMetricName = PrometheusNaming.sanitizeLabelName(metricName);
+        Labels newLabels = labels;
+        if (labels.contains(sanitizedMetricName)) {
+            LOG.warning("Ignoring metric value duplicate key: " + sanitizedMetricName);
```

Comment:
It's probably worth print a bit more information such as the metric name so operators can identify the culprit.

Reply:
Also we should not use concatenation in log lines but instead use interpolation with the `{}` syntax which is possible with the slf4j `Logger`

Reply:
I appended `from metric; metricName`. Is that sufficient. 
`{}` interpolation now used